folio-spring-cql/src/main/java/org/folio/spring/cql/Cql2JpaCriteria.java

```diff
@@ -298,28 +298,40 @@ private Predicate indexNode(Path<?> field, CQLTermNode node, CqlModifiers modifi
       return buildModifiersQuery(field, modifiers, cb);
     }
 
-    var isString = String.class.equals(field.getJavaType());
     var comparator = node.getRelation().getBase().toLowerCase();
     var term = node.getTerm();
 
+    if (equalsAny(comparator, "=", "==") && (StringUtils.isBlank(term) || term.equals("''"))) {
```

Comment:
```suggestion
    if (equalsAny(comparator, "=", "==") && StringUtils.isEmpty(term)) {
```
@mukhiddin-yusuf, `''` was my mistake in changes that I shared with you
And I think we have to check only for empty, to not touch queries like `field1="    "`